src/mode/modeHandler.ts

```diff
@@ -1510,7 +1510,17 @@ export class ModeHandler implements vscode.Disposable {
       opCursorCharDecorations
     );
 
-    // TODO: draw marks (#3963)
+    if (configuration.showMarksInGutter) {
+      for (const { position, name } of this.vimState.historyTracker.getMarks()) {
+        const markLine = position.getLineBegin();
+        const markRange = new vscode.Range(markLine, markLine);
+        const markDecoration = decoration.marks[name];
+
+        if (this.vimState.editor.hasOwnProperty('setDecorations')) {
```

Comment:
Why are you checking whether `editor.setDecorations` exists? We used it just above, so if it doesn't, we've presumably already thrown. Is there any reason to suspect it may not exist?

Reply:
This check actually fails for me (Maybe `TextEditor` is a subclass of whatever defines `setDecorations`?) so the feature doesn't work unless I remove it

Reply:
When I ran the test suite without this condition, I ran into a bunch of errors about `editor` not having the key `setDecoration`. I'm still fairly unfamiliar with the project and I wasn't quite sure what was going on. I can try to dig into it further.

Reply:
Just pushed up fixes to the other feedback you provided (thank you!).

We'll see if the tests pass with those updates 🤞